src/test/kotlin/open/tresorier/services/OperationServiceTest.kt

```diff
@@ -63,4 +63,87 @@ class OperationServiceTest : ITest {
         Assertions.assertTrue(orderInDayAtCreation < operationModified.orderInDay)
     }
 
-}
\ No newline at end of file
+    @Test
+    fun testOfxOperationDebit() {
+        val rosalind: Person = personService.createPerson(
+            "Rosalind Franklin", "Cliché_51", "rosalind@adn.uk", ProfileEnum.PROFILE_USER
+        )
+        rosalind.billingStatus = true
+        val budget: Budget = budgetService.findByUser(rosalind)[0]
+        val account: Account = accountService.create(
+            rosalind, budget, "personal account", TestData.jan_14_2022, 1000
+        )
+        val ofxDebitOperation: String = "<STMTTRN><TRNTYPE>DEBIT<DTPOSTED>20220114<TRNAMT>-536,50<FITID>2802202220220228-03.01.56.182263<NAME>PRIX LOUISA GROSS HORWITZ<MEMO>PRIX LOUISA GROSS HORWITZ</STMTTRN>"
+
+        val newOperation: Operation = operationService.createOperationFromOFX(rosalind, account, ofxDebitOperation)
+
+        Assertions.assertTrue(newOperation.day.comparable == 20220114 && newOperation.amount == -53650 && newOperation.memo == "PRIX LOUISA GROSS HORWITZ")
```

Comment:
you could create an assert for each to help debug. that way the error would happen on the line that is problematic instead of just on the whole condition.
It may also be useful to create a isEqual function in operation so that operation.isEquals(otherOperation) cna be easily called everywhere

Reply:
done